lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -285,11 +392,119 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         this.categoryControlVisible = false;
     }
 
-    cancelChanges() {
+    cancelChanges(buttonType: ButtonType, event: MouseEvent, group?: CardViewGroup) {
+        event.stopPropagation();
+        this.toggleEditMode(buttonType, group);
         this.revertChanges();
         this.loadProperties(this.node);
     }
 
+    // Returns the editing state of the panel
+    isEditingPanel(): boolean {
+        return (
+            (this.editable && this.hasMetadataChanged) ||
+            (this.editableTags && this.hasMetadataChanged) ||
+            (this.editableCategories && this.hasMetadataChanged) ||
+            (this.editableGroup?.editable && this.hasMetadataChanged)
+        );
+    }
+
+    toggleEdit(event: MouseEvent, group: CardViewGroup, buttonType: ButtonType): void {
+        event.stopPropagation();
+        if (this.isEditingPanel()) {
+            this.notificationService.showError('METADATA.BASIC.SAVE_OR_DISCARD_CHANGES');
+            return;
+        }
+
+        if (this.editableGroup && this.editableGroup !== group) {
```

Comment:
You comparing if the object is the same, is this valid? Maybe you should compare by panel title?

Reply:
Compared with the group title.